extensions/kind/src/extension.ts

```diff
@@ -80,22 +80,51 @@ async function updateClusters(provider: extensionApi.Provider, containers: exten
       status,
       apiPort: listeningPort?.PublicPort || 0,
       engineType: container.engineType,
+      engineId: container.engineId,
+      id: container.Id,
+    };
+  });
+  kindClusters = kindContainers.map(container => {
+    return {
+      name: container.name,
+      status: container.status,
+      apiPort: container.apiPort,
+      engineType: container.engineType,
     };
   });
 
-  kindClusters.forEach(cluster => {
+  kindContainers.forEach(cluster => {
     const item = registeredKubernetesConnections.find(item => item.connection.name === cluster.name);
     const status = () => {
       return cluster.status;
     };
     if (!item) {
+      const lifecycle: extensionApi.ProviderConnectionLifecycle = {
+        start: async (): Promise<void> => {
+          try {
+            // start the container
+            await extensionApi.containerEngine.startContainer(cluster.engineId, cluster.id);
```

Comment:
it can't be containerEngine. We register kube objects

Kubernetes connections are different than containers